pkg/apis/apps/validation/validation_test.go

```diff
@@ -3070,6 +3103,209 @@ func TestValidateDeploymentStatusUpdate(t *testing.T) {
+	}
+	successCases := []depUpdateTest{
+		{
+			old: apps.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+				Spec: apps.DeploymentSpec{
+					Selector: &metav1.LabelSelector{MatchLabels: validLabels},
+					Template: validPodTemplate.Template,
+					Strategy: apps.DeploymentStrategy{Type: apps.RecreateDeploymentStrategyType},
+				},
+			},
+			update: apps.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+				Spec: apps.DeploymentSpec{
+					Replicas: 1,
+					Selector: &metav1.LabelSelector{MatchLabels: validLabels},
+					Template: readWriteVolumePodTemplate.Template,
+					Strategy: apps.DeploymentStrategy{Type: apps.RecreateDeploymentStrategyType},
+				},
+			},
+		},
+		{
+			enableSkipReadOnlyValidationGCE: true,
+			old: apps.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+				Spec: apps.DeploymentSpec{
+					Selector: &metav1.LabelSelector{MatchLabels: validLabels},
+					Template: validPodTemplate.Template,
+					Strategy: apps.DeploymentStrategy{Type: apps.RecreateDeploymentStrategyType},
+				},
+			},
+			update: apps.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+				Spec: apps.DeploymentSpec{
+					Replicas: 1,
+					Selector: &metav1.LabelSelector{MatchLabels: validLabels},
+					Template: readWriteVolumePodTemplate.Template,
+					Strategy: apps.DeploymentStrategy{Type: apps.RecreateDeploymentStrategyType},
+				},
+			},
+		},
+	}
+	for _, successCase := range successCases {
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SkipReadOnlyValidationGCE, successCase.enableSkipReadOnlyValidationGCE)()
+		successCase.old.ObjectMeta.ResourceVersion = "1"
+		successCase.update.ObjectMeta.ResourceVersion = "1"
+		if errs := ValidateDeploymentUpdate(&successCase.update, &successCase.old, corevalidation.PodValidationOptions{}); len(errs) != 0 {
+			t.Errorf("expected success: %v", errs)
+		}
+	}
+	errorCases := map[string]depUpdateTest{
+		"more than one read/write": {
+			old: apps.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "", Namespace: metav1.NamespaceDefault},
+				Spec: apps.DeploymentSpec{
+					Selector: &metav1.LabelSelector{MatchLabels: validLabels},
+					Template: validPodTemplate.Template,
+					Strategy: apps.DeploymentStrategy{Type: apps.RecreateDeploymentStrategyType},
+				},
+			},
+			update: apps.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+				Spec: apps.DeploymentSpec{
+					Replicas: 2,
+					Selector: &metav1.LabelSelector{MatchLabels: validLabels},
+					Template: readWriteVolumePodTemplate.Template,
+					Strategy: apps.DeploymentStrategy{Type: apps.RecreateDeploymentStrategyType},
+				},
+			},
+		},
+		"invalid read-write volume": {
+			enableSkipReadOnlyValidationGCE: false,
```

Comment:
You should either always have this line in each case, or ONLY have this line when setting to true.